src/config/configuration.validator.ts

```diff
@@ -0,0 +1,23 @@
+import Ajv, { Schema } from 'ajv';
+
+const configurationSchema: Schema = {
+  type: 'object',
+  properties: {
+    AUTH_TOKEN: { type: 'string' },
+    EXCHANGE_API_KEY: { type: 'string' },
+  },
+  required: ['AUTH_TOKEN', 'EXCHANGE_API_KEY'],
+};
+
+export function validate(
+  configuration: Record<string, unknown>,
+): Record<string, unknown> {
+  if (process.env.NODE_ENV !== 'test') {
+    const ajv = new Ajv({ allErrors: true });
```

Comment:
Here we need to export a function, so we can't use DI to inject the global AJV instances and the common validation utilities we have in place. So I opted to stay simple a just create a new AJV instance. This function will be executed only once during the application bootstrap, and the instance will be immediately garbage-collected, so I tried to keep it simple in this case.